anitya/lib/backends/github.py

```diff
@@ -50,27 +59,167 @@ def get_versions(cls, project):
+            url = project.version_url
         elif project.homepage.startswith('https://github.com'):
-            url = project.homepage
+            url = project.homepage.replace('https://github.com/', '')
             if url.endswith('/'):
-                url = project.homepage[:-1]
-            url += '/tags'
+                url = url[:-1]
         else:
             raise AnityaPluginException(
                 'Project %s was incorrectly set-up' % project.name)
 
-        return get_versions_by_regex(url, REGEX, project)
+        try:
+            (owner, repo) = url.split('/')
+        except ValueError:
+            raise AnityaPluginException(
+                'Project %s was incorrectly set-up. \
+                Can\'t parse owner and repo' % project.name)
+
+        query = prepare_query(owner, repo)
+
+        try:
+            headers = REQUEST_HEADERS.copy()
+            token = config.get('GITHUB_ACCESS_TOKEN')
+            if token:
+                headers['Authorization'] = 'bearer %s' % token
+            resp = http_session.post(
+                API_URL, json={'query': query}, headers=headers, timeout=60, verify=True)
+        except Exception as err:
+            _log.debug('%s ERROR: %s' % (project.name, str(err)))
+            raise AnityaPluginException(
+                'Could not call : "%s" of "%s", with error: %s' % (
+                    API_URL, project.name, str(err)))
+
+        if resp.ok:
+            json = resp.json()
+        else:
+            raise AnityaPluginException(
+                '%s: Server responded with status "%s": "%s"' % (
+                    project.name, resp.status_code, resp.reason))
+
+        versions = parse_json(json, project)
+
+        return versions
+
+
+def parse_json(json, project):
+    ''' Function for parsing json response
+
+    Attributes:
+        json (dict): Json dictionary to parse.
+        project (:obj:`anitya.db.models.Project`): Project object whose backend
+            corresponds to the current plugin.
+
+    Returns:
+        :obj:`list`: A list of all the possible releases found.
+
+    Raises:
+        AnityaPluginException: A
+            :obj:`anitya.lib.exceptions.AnityaPluginException` exception
+            when the versions cannot be retrieved correctly.
+        RateLimitException: A
+            :obj:`anitya.lib.exceptions.RateLimitException` exception
+            when rate limit is reached.
+
+    '''
+    if 'errors' in json:
+        error_str = ''
+        for error in json['errors']:
+            error_str += '"%s": "%s"\n' % (
+                error['type'], error['message'])
+        raise AnityaPluginException(
+            '%s: Server responded with following errors\n%s' % (
+                project.name, error_str))
+
+    total_count = json['data']['repository']['refs']['totalCount']
+
+    _log.debug('Received %s tags for %s' % (total_count, project.name))
+
+    # TODO: Save cursor to project and use it to fetch only newer versions
```

Comment:
I recommend turning these TODOs into tickets and removing them from the code

Reply:
Will do